ledger/participant-integration-api/src/main/scala/platform/apiserver/configuration/LedgerConfigProvisioner.scala

```diff
@@ -17,92 +16,86 @@ import com.daml.platform.configuration.LedgerConfiguration
-    timeProvider: TimeProvider,
-    submissionIdGenerator: SubmissionIdGenerator,
-    config: LedgerConfiguration,
-    materializer: Materializer,
-)(implicit loggingContext: LoggingContext)
-    extends AutoCloseable {
+object LedgerConfigProvisioner {
   private val logger = ContextualizedLogger.get(getClass)
 
-  private val closed: AtomicBoolean = new AtomicBoolean(false)
-
-  materializer.scheduleOnce(
-    config.initialConfigurationSubmitDelay.toNanos.nanos,
-    () => {
-      if (currentLedgerConfiguration.latestConfiguration.isEmpty && !closed.get)
-        submitInitialConfig(writeService)
-      ()
-    },
-  )
+  def owner(
+      ledgerConfiguration: LedgerConfiguration,
+      currentLedgerConfiguration: CurrentLedgerConfiguration,
+      writeService: state.WriteConfigService,
+      timeProvider: TimeProvider,
+      submissionIdGenerator: SubmissionIdGenerator,
+      scheduler: Scheduler,
+      servicesExecutionContext: ExecutionContext,
+  )(implicit
+      loggingContext: LoggingContext
+  ): ResourceOwner[Unit] = {
+    ResourceOwner
+      .forCancellable(() =>
+        scheduler.scheduleOnce(
+          ledgerConfiguration.initialConfigurationSubmitDelay.toNanos.nanos,
+          new Runnable {
+            override def run(): Unit = {
+              if (currentLedgerConfiguration.latestConfiguration.isEmpty)
+                submitInitialConfig(writeService, timeProvider, submissionIdGenerator)(
+                  ledgerConfiguration.initialConfiguration
+                )(servicesExecutionContext, loggingContext)
+              ()
+            }
+          },
+        )(servicesExecutionContext)
+      )
+      .map(_ => ())
+  }
 
   // There are several reasons why the change could be rejected:
   // - The participant is not authorized to set the configuration
   // - There already is a configuration, it just didn't appear in the index yet
   // This method therefore does not try to re-submit the initial configuration in case of failure.
-  private def submitInitialConfig(writeService: state.WriteConfigService): Future[Unit] = {
-    val submissionId = submissionIdGenerator.generate()
-    withEnrichedLoggingContext("submissionId" -> submissionId) { implicit loggingContext =>
-      logger.info(s"No ledger configuration found, submitting an initial configuration.")
-      DefaultTelemetry.runFutureInSpan(
-        SpanName.LedgerConfigProviderInitialConfig,
-        SpanKind.Internal,
-      ) { implicit telemetryContext =>
-        FutureConverters
-          .toScala(
-            writeService.submitConfiguration(
-              Timestamp.assertFromInstant(timeProvider.getCurrentTime.plusSeconds(60)),
-              submissionId,
-              config.initialConfiguration,
+  private def submitInitialConfig(
+      writeService: state.WriteConfigService,
+      timeProvider: TimeProvider,
+      submissionIdGenerator: SubmissionIdGenerator,
+  )(
+      initialConfiguration: Configuration
```

Comment:
It doesn't look like you're using this curry

Reply:
No, but I like that the "data" parameters are separate from the "collaborator" arguments. Not sure of a better way to do this; I guess I could extract out a class.

Reply:
I agree this intention is better conveyed by adding a bit more structure rather than through separate parameter lists.

Reply:
OK, sounds good, I shall do that.

Reply:
Handled in https://github.com/digital-asset/daml/pull/10461/commits/45734de114b8c355355f1910ae896fb41d044a5f.